src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
+    }
 
     @Override
     public void put(K key, V value) {
-
+        growIfHashMapFull();
+        int index = getValidIndex(key, table.length);
+        Node<K, V> currentNode = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (currentNode == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> nodeToUpdate = updateIfValidKey(currentNode, key, value);
+            if (nodeToUpdate != null) {
+                nodeToUpdate.next = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getValidIndex(key, table.length);
+        Node<K, V> node = getNode(index);
+        if (node != null) {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void growIfHashMapFull() {
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
+        }
+    }
+
+    private int getValidIndex(K key, int capacity) {
+        return (capacity - 1) & hash(key,capacity);
+    }
+
+    private int hash(K key, int capacity) {
+        return (key == null) ? 0 : key.hashCode() % capacity;
+    }
+
+    private Node<K, V> updateIfValidKey(Node<K, V> node, K key, V value) {
+        while (node.next != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return null;
+            }
+            node = node.next;
+        }
+        if (Objects.equals(key, node.key)) {
+            node.value = value;
+            return null;
+        }
+        return node;
+    }
+
+    private Node<K, V> getNode(int index) {
+        return table[index] != null ? table[index] : null;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * SIZE_INCREASE_FACTOR;
+        Node<K, V>[] newTable = new Node[newCapacity];
+        for (Node<K, V> node : table) {
+            if (node != null) {
+                moveToNewTable(node, newTable);
+            }
+        }
+        table = newTable;
+    }
+
+    private void moveToNewTable(Node<K, V> oldNode, Node<K, V>[] newTable) {
+        while (oldNode != null) {
+            int index = getValidIndex(oldNode.key, newTable.length);
+            Node<K, V> newNodePosition = newTable[index];
+            Node<K, V> newNode = new Node<>(oldNode.key, oldNode.value, null);
+            if (newNodePosition == null) {
+                newTable[index] = newNode;
+            } else {
+                Node<K, V> lastNode = getLastNode(newNodePosition);
+                lastNode.next = newNode;
+            }
+            oldNode = oldNode.next;
+        }
```

Comment:
over engeeniring happend. Just create a new table and use  put(K key, V value) while iterating all nodes in the old table